lib/CMakeLists.txt

```diff
@@ -98,9 +100,30 @@ add_library(LLVMdg SHARED
 	llvm/analysis/DefUse/DefUse.cpp
 )
 
+# Get proper shared-library behavior (where symbols are not necessarily
+# resolved when the shared library is linked) on OS X.
+if(APPLE)
+  set_target_properties(LLVMdg PROPERTIES
+    LINK_FLAGS "-undefined dynamic_lookup"
+  )
+
+  set_target_properties(LLVMpta PROPERTIES
+    LINK_FLAGS "-undefined dynamic_lookup"
+  )
+
+  set_target_properties(LLVMrd PROPERTIES
+    LINK_FLAGS "-undefined dynamic_lookup"
+  )
+endif(APPLE)
+
 target_link_libraries(LLVMdg
 			PUBLIC LLVMpta
-			PUBLIC LLVMrd)
+			PUBLIC LLVMrd
+			PRIVATE ${llvm_support}
```

Comment:
Could you make this change also dependent on APPLE? Projects linking to dg on Linux may have troubles when LLVM is already linked because of the CommandLine static variables.